src/golang/lib/k8s/clients.go

```diff
@@ -0,0 +1,85 @@
+package k8s
+
+import (
+	"context"
+
+	"github.com/dropbox/godropbox/errors"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	// This key is used to create a label for every pod that gives the name of
+	// the service that it is running. For user workloads, the value will be
+	// the user-specified service name.
+	ServiceKey = "serviceName"
+)
+
+// This is a helper function that creates a Kubernetes Clientset using the
+// `client-go/rest` library's `InClusterConfig()` function. This function will
+// only succeed if it is called from within a Kubernetes cluster. Otherwise,
+// the configuration retrieval will fail, and this function will call
+// `log.Fatal` and cause the program to crash. If this function is called from
+// within the cluster, it should not fail.
+func CreateClientInCluster() *kubernetes.Clientset {
+	k8sConfig, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatalf("Unexpected error while creating Kubernetes client: %v\n", err)
+	}
+
+	k8sClient, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		log.Fatalf("Unexpected error while creating Kubernetes client: %v\n", err)
+	}
+
+	return k8sClient
+}
+
+// This is a helper function that creates a Kubernetes Clientset using the
+// kubeconfig that is located at `kubecfgLocation`. By default, this location
+// is specified (in `enterprise/config/cluster.yml`) as `~/.kube/config`. Please modify
+// the `cluster.yml` file if you need to change this location. If this
+// location is misconfigured, this function will fail.
+func CreateClientOutsideCluster(kubecfgLocation string) (*kubernetes.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubecfgLocation)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unexpected error while creating Kubernetes client.")
+	}
+
+	k8sClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, errors.Wrap(err, "Unexpected error while creating Kubernetes client.")
+	}
+
+	return k8sClient, nil
+}
+
+// This is a helper function that creates the user namespace
+// and does not return anything. This function should never
+// fail, so any errors that are encountered call `log.Fatal` and cause the
+// program to crash.
+func CreateNamespaces(k8sClient *kubernetes.Clientset) error {
```

Comment:
Should we call the namespace we use something else? Like `aqueduct` to try to ensure that there are no conflicts with existing namespaces the customer has created.